subprojects/gradle/build-checks/src/main/kotlin/com/avito/android/build_checks/BuildParamCheckPlugin.kt

```diff
@@ -58,7 +57,8 @@ public open class BuildParamCheckPlugin : Plugin<Project> {
 
         if (!project.pluginIsEnabled) return
 
-        val logger = GradleLoggerFactory.getLogger(this, project)
+        val loggerFactory = GradleLoggerFactory.fromProject(project)
+        val logger = loggerFactory.create(this::class.java.simpleName)
```

Comment:
Now clients have to do these two operations instead of one request.
How about hiding it inside factory?

Reply:
done in https://github.com/avito-tech/avito-android/pull/1198